lms/djangoapps/courseware/access.py

```diff
@@ -172,6 +180,24 @@ def has_access(user, action, obj, course_key=None):
                     .format(type(obj)))
 
 
+def _is_banned_from_course(user, course_key):
+    """
+    Checks if the user is banned from the course its trying to access
+    """
+    if AllCoursesEnrollmentBanned.objects.filter(email=user.email, is_active=1).exists():
+        return True
+
+    banned_user = CourseEnrollmentBanned.objects.filter(email=user.email, course_id=course_key, is_active=1).first()
+
+    if not banned_user:
+        return False
+
+    if banned_user.expiry and banned_user.expiry < datetime.now(UTC):
```

Comment:
Can we add a instance method or property  in the class named `is_expired` instead of comparing dates here? what do you think?